### PR TITLE
fix crash in multi-modal

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sywangyi** [comment]
@Narsil please have review. find the issue when enable this model in xpu. should be common issue.

**Comment by sywangyi** [comment]
@yao-matrix @yuanwu2017

File: server/text_generation_server/models/custom_modeling/flash_pali_gemma_modeling.py [hunk 1]
```diff
@@ -92,7 +93,7 @@ def forward(
 
             # insert image features into input embeddings
             inputs_embeds[mask] = image_features.view(-1, image_features.shape[-1])
-
+        input_lengths = Seqlen(input_lengths=input_lengths)
```
Comment (danieldk) [inline server/text_generation_server/models/custom_modeling/flash_pali_gemma_modeling.py:96]: This currently breaks CUDA inference.

I haven't looked deeply yet, but the issue could be that the wrapping should not happen with CUDA Graphs enabled (see the flash causal LM implementation).

Also, I think we could do the wrapping in `VlmCausalLM.forward`, which would be nicer than doing it per model.

Comment (sywangyi) [reply server/text_generation_server/models/custom_modeling/flash_pali_gemma_modeling.py]: agreed,update the PR

**Comment by sywangyi** [comment]
@OlivierDehaene to be aware of the fix in  https://github.com/huggingface/text-generation-inference/pull/2245/commits/190368e1377224d654bc15c77d5b96ca5d295052

**Comment by danieldk** [comment]
Thanks a lot for the changes! Rerunning tests now.

**Pull request merged**